js/data/bucket/symbol_bucket.js

```diff
@@ -218,16 +218,28 @@ SymbolBucket.prototype.populateArrays = function(collisionTile, stacks, icons) {
         textFeatures = merged.textFeatures;
     }
 
-    var shapedText, shapedIcon;
+    var shapedText, shapedTextVertical, shapedIcon;
 
     for (var k = 0; k < features.length; k++) {
         if (!geometries[k]) continue;
 
+        var verticalOrientation = false;
+
         if (textFeatures[k]) {
+
             shapedText = shapeText(textFeatures[k], stacks[fontstack], maxWidth,
-                    lineHeight, horizontalAlign, verticalAlign, justify, spacing, textOffset);
+                    lineHeight, horizontalAlign, verticalAlign, justify, spacing, textOffset, oneEm, verticalOrientation);
+
+            if (layout['text-rotation-alignment'] === 'map' && layout['symbol-placement'] === 'line') {
```

Comment:
Note that since we switched to Buble, we can now use `u` `RegExp` flag that allows using any unicode characters in regexps (they get transpiled to `\u....`).